tests/samples/expected/ocaml.ml

```diff
@@ -222,7 +222,8 @@ let add_channel b ic len =
 let output_buffer oc b =
   output oc b.buffer 0 b.position
 
-let closing = function
+let closing =
```

Comment:
On this one, I'd like to ask @Niols and @aspiwack for advice.

Reply:
Are you familiar with [George Orwell's _A Nice Cup of Tea_](https://www.orwellfoundation.com/the-orwell-foundation/orwell/essays-and-other-works/a-nice-cup-of-tea/) essay? I promise that it's on topic.

Anyway, I would keep `function` trailing as in

```ocaml
let closing = function
  | '(' -> ')'
  | '{' -> '}'
  | _ -> assert false
```

(this is the same as before change)

There are few such hanging constructions, I can think of the more complicated

```ocaml
let foo : int -> int -> int = fun x y ->
  fibonacci_is_a_long_name_to_write x + isn't_it_yes_this_is_a_legal_function_name y
````

I usually hang `match` myself, but this looks like a rarer style (so we should probably not go this route):

```ocaml
let bar x y = match x,y with
  | true, false -> "banana"
  | false, true -> "apple"
  | _, _ -> fail_with "I can't think of another fruit"
```

It's probably all I would leave hanging like this.

Reply:
> Are you familiar with [George Orwell's _A Nice Cup of Tea_](https://www.orwellfoundation.com/the-orwell-foundation/orwell/essays-and-other-works/a-nice-cup-of-tea/) essay? I promise that it's on topic.

So is that a hint that I am being pedantic?

By the way, I find it amusing that the British can argue about whether or not to put sugar in tea, but when it comes to milk, the notion of not using it (in tea or coffee) seems completely alien to them.

Reply:
Since a `match` can contain arbitrary terms, especially the ones for which the formatter prints a like break, I think that putting it on the line of the equal will lead to quite unaesthetic (or inconsistent) result, so I am in favour of breaking the line before it.
For `fun` and `function`, I have no opinion about it. When I manually format, I tend to break line before it, but this is not among the things I am ready to fight for.

Reply:
The reason I asked was that I was unsure if there is an established OCaml style about this. But it sounds like perhaps not.

Reply:
@torhovland 
> So is that a hint that I am being pedantic?

Not at all. I'm saying that a lot of questions regarding formatting have one obvious unquestionable answer to me. And that all these answers are wrong for someone else.

@GuillaumeGen 
> For fun and function, I have no opinion about it. When I manually format, I tend to break line before it, but this is not among the things I am ready to fight for.

Honestly, I don't know what other people do. I'll ask.

Reply:
For `function`, personally, I'd go for:
```ocaml
let closing = function
  | '(' -> ')'
  | '{' -> '}'
  | _ -> assert false
```
and I feel this one is fairly common.

---

For `fun`, I would sometimes do:
```ocaml
let foo : int -> int -> int = fun x y ->
  fibonacci_is_a_long_name_to_write x + isn't_it_yes_this_is_a_legal_function_name y
```
but I think more generally I'd put the `fun` on the next line here; might not be the popular version though:
```ocaml
let foo : int -> int -> int =
  fun x y ->
  fibonacci_is_a_long_name_to_write x + isn't_it_yes_this_is_a_legal_function_name y
```
I don't think there is a clearly established style for that one. If there is one, I'd say it's maybe the first one? Although then I wonder what's best in the case where there is both arguments before the `=` and after the `fun`, eg.:
```ocaml
let foo (x : int) : int -> int = fun y ->
  fibonacci_is_a_long_name_to_write x + isn't_it_yes_this_is_a_legal_function_name y
```
Although actually equivalent, I think the last one really means “`foo` is a function of one argument `x : int` that then returns a function from `int` to `int` defined as `fun y -> ...`”, while the others mean “`foo` is a function of two arguments `x : int` and `y : int`”. In the last case, I do think putting `fun y ->` on a new line brings more clarity.

---

Finally, I tend to put `match` on its own line.
```ocaml
let bar x y =
  match x,y with
  | true, false -> "banana"
  | false, true -> "apple"
  | _, _ -> fail_with "I can't think of another fruit"
```
although I have occasionally done Arnaud's version as well for very short things (like in the given example).

Reply:
I've got one vote today, so I'll share before I say goodnight: this first vote would let both `function` and `fun` hang (honestly, I'm the only person in the world hanging `match`, so I'm not even asking).